### DISCUSSION TITLE
Making dons and kats look different and the slippery slope to taikomania hell

### DISCUSSION THREAD

**Discussion by quat1024** [ppy/osu#21105] [Ideas]
What is "taikomania"?

> it’s a skin that converts the usual single lane taiko playfield into a dual lane playfield with each side belonging to either color that makes reading much easier for mania players (and sometimes non-mania players aswell). however, because it’s not the intended way to play taiko (as it basically makes it mania-like), it receives very constant backlash by the community
> 
> it basically allows mania players to completely tear the leaderboards apart. [---](https://www.reddit.com/r/osugame/comments/f92awz/taikomania_skin_patched_on_cutting_edge/fir4udp/)

The original skin thread is still up: https://osu.ppy.sh/community/forums/topics/404353

Basically osu!stable uses the same png for dons and kats and tints it at runtime. In the past, you could counteract the game's tinting by including colors in the image, so that regions of the image would turn black and you were able to create two different shapes for dons and kats. For example, you could separate all the dons into one column and kats into another, and then play taiko just like it's a sideways version of 2k/4k osu!mania. If you're better at reading column-scrolling rhythm games than you are at reading colors this was an appealing way to unfairly dominate the leaderboards.

People did get banned over this. It was fixed by forcing the image to grayscale when loading it. It's also an item on the [things to do for anticheat](https://github.com/ppy/osu/issues/18876) list. This was good, i'm not proposing allowing this again, this should absolutely be banned on the leaderboards.

But......!

There's interest in skinning dons and kats differently, even for non nefarious purposes. [Adding don and kat text under notes](https://github.com/ppy/osu/discussions/13249) is an oft-suggested thing that the arcade Taiko games do. The Lazer *default skin* "circles!" even styles dons with a filled-in white circle and kats with a hollow circle. It is a slippery slope between this stuff and "taikomania", right? If you allow adding `Don` and `Kat` text under notes, you also allow adding `‾` and `_` text under notes and oops all taikomania.

What about in an unranked context? I feel like osu!lazer has more room for features that cater to casual players? Taikomania was fixed in stable because it affected competitive gameplay, but Lazer has all these mods like Magnetized which are just a good amount of fun. One of my friends mentioned using the taikomania skin to casually experience taiko, just to have fun with the music, not to climb the leaderboards. I don't think people would be nearly as upset about taikomania skins if they were somehow unranked, or came with a hefty score/pp penalty, or something.

## things to think about

* **Pretend this never happened** - Greyscale the taiko hitcircle image, change "circles!" to use the same styling for dons and kats, explicitly deny the request to add text under taiko objects, do not pass go, do not collect $200.
* **A taikomania style mod** - *unranked* mod that vertically moves dons and kats apart. Players can goof off with it and have their silly fun but it wouldn't get on leaderboards.
* **Autoconversion from Taiko charts to 2k/4k Mania charts** - Taikomania skin users basically just wanted to play mania anyway, let them experience the library of taiko charts without the middleman

**Comment by smoogipoo** [comment]
> Players can goof off with it and have their silly fun but it wouldn't get on leaderboards.

In lazer, all mods are ranked (with the exception of autoplay).

> Autoconversion from Taiko charts to 2k/4k Mania charts

Fyi, I'm fairly certain osu!lazer's mania already supports this to a degree. Also mania -> taiko :) It's just that song select prevents this for now.

> taikomania style mod

I generally disagree with this because having a mod exist for a ruleset to become another ruleset sounds super weird to me.

**Reply by Saraistupid** [reply]
> Fyi, I'm fairly certain osu!lazer's mania already supports this to a degree. Also mania -> taiko :) It's just that song select prevents this for now.

Sorry for necrobumping, but this is really interesting, is there a reason for why that's not enabled yet? Is it a balancing issue or just lack of interest?

**Reply by peppy** [reply]
It's newly added complexity we are looking to avoid.
